moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -149,16 +153,53 @@ ServoCalcs::ServoCalcs(rclcpp::Node::SharedPtr node, const ServoParametersPtr& p
   {
     // A map for the indices of incoming joint commands
     joint_state_name_map_[internal_joint_state_.name[i]] = i;
-
-    // Low-pass filters for the joint positions
-    position_filters_.emplace_back(parameters_->low_pass_filter_coeff);
   }
 
   // A matrix of all zeros is used to check whether matrices have been initialized
   Eigen::Matrix3d empty_matrix;
   empty_matrix.setZero();
   tf_moveit_to_ee_frame_ = empty_matrix;
   tf_moveit_to_robot_cmd_frame_ = empty_matrix;
+
+  // Reflexxes for command smoothing
+  reflexxes_ = std::make_unique<ReflexxesAPI>(num_joints_, parameters_->publish_period, 0 /* additional threads */);
+  reflexxes_position_input_param_ = std::make_unique<RMLPositionInputParameters>(num_joints_);
+  reflexxes_position_output_param_ = std::make_shared<RMLPositionOutputParameters>(num_joints_);
+  // All joints are active
+  reflexxes_wrapper::setSelectionVectorAllTrue(reflexxes_position_input_param_.get(), num_joints_);
+  // Initialize positions/vels/accels to zeros. Current positions will be updated later
+  reflexxes_wrapper::initializePositionInputStateToZeros(reflexxes_position_input_param_.get(), num_joints_);
```

Comment:
Is it safe to initialize the position vector to 0? Position should correspond to the current state of the robot. In ros2_control we are initializing positions to `std::numeric_limits<double>::quiet_NaN()` in constructors and then when the robot states are updated to real values assign those. Using this approach we can distinguish if data are initialized from the hardware or only variable initialization is done. Otherwise, unintended movement toward positions 0 could happen. (May is this protected with another variable, if so, ignore this comment).

Reply:
good idea - done!